src/modules/chat/qbChat.js

```diff
@@ -664,19 +666,39 @@ ChatProxy.prototype = {
         Utils.QBLog('[Chat]', 'Connect with parameters ' + JSON.stringify(params));
 
         var self = this,
-            err, rooms;
+            err,
+            rooms;
 
         var userJid = chatUtils.buildUserJid(params);
 
+        if (self._isConnecting) {
+            return;
```

Comment:
I'm not sure that this is ideal because a user expects the callback to be fired, but with your fix nothing will happen.